frontend/src/metabase/css/core/z-index.module.css

```diff
@@ -1,3 +1,13 @@
+:root {
+  --mb-floating-element-z-index: 200;
+}
+
+.FloatingElement {
+  &:not([data-specificity-hack]) {
+    z-index: var(--mb-floating-element-z-index);
+  }
+}
```

Comment:
vocab-wise, I'm torn between 'overlay' (shorter, canonical but collides with the other usage of Overlay to mean 'translucent backdrop') and 'floating element' (longer, though accurate. We do put Modal, Menu, etc. in an 'overlays' directory fwiw